src/main/java/org/opentripplanner/updater/trip/TimetableSnapshotSource.java

```diff
@@ -689,34 +706,111 @@ private Result<?, UpdateError> handleAddedTrip(
 
     return addTripToGraphAndBuffer(
       tripBuilder.build(),
-      tripUpdate,
+      tripUpdate.getVehicle(),
+      stopTimeUpdates,
       stops,
       serviceDate,
       RealTimeState.ADDED
     );
   }
 
+  private Route getOrCreateRoute(TripDescriptor tripDescriptor, FeedScopedId tripId) {
+    if (routeExists(tripId.getFeedId(), tripDescriptor)) {
+      // Try to find route
+      return transitService.getRouteForId(
+        new FeedScopedId(tripId.getFeedId(), tripDescriptor.getRouteId())
+      );
+    }
+    // the route in this update doesn't already exist, but the update contains the information so it will be created
+    else if (
+      tripDescriptor.hasExtension(MfdzRealtimeExtensions.tripDescriptor) &&
+      !routeExists(tripId.getFeedId(), tripDescriptor)
```

Comment:
`routeExists(tripId.getFeedId(), tripDescriptor)` is already checked on the previous if, so no need to check it a second time. There is still some shared code between the two blocks, but we can refactor it at a later stage